src/librustc/hir/mod.rs

```diff
@@ -1426,7 +1426,7 @@ pub enum Ty_ {
     /// A reference (`&'a T` or `&'a mut T`)
     TyRptr(Lifetime, MutTy),
     /// A bare function (e.g. `fn(usize) -> bool`)
-    TyBareFn(P<BareFnTy>),
+    TyBareFn(P<BareFnTy>, HirVec<Spanned<Name>>),
```

Comment:
When we discussed it, you were supposed to put it in `BareFnTy`. That should simplify things.

Reply:
Ah damn, got tricked by [this message](https://botbot.me/mozilla/rust-internals/2017-09-26/?msg=91596420&page=5).